frontend/src/components/common/Typography.tsx

```diff
@@ -0,0 +1,49 @@
+import { PropsWithChildren, useState } from 'react'
+
+import { Typography as MaterialTypography, Theme, TypographyProps } from '@material-ui/core'
+import { createStyles, makeStyles } from '@material-ui/core/styles'
+import LinkIcon from '@material-ui/icons/Link'
+
+import LinkIconButton from 'components/common/LinkIconButton'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    linkIcon: {
+      transform: 'rotate(-45deg)',
+      color: theme.palette.primary.main,
+      '&:hover': {
+        color: theme.palette.primary.dark,
+      },
+    },
+    hideLinkIcon: {
+      visibility: 'hidden',
+    },
+  }),
+)
+
+type TypographyParams = PropsWithChildren<TypographyProps> & {
+  id?: string
+  linkable?: boolean
+  component?: string
+}
```

Comment:
Here's my proposal for these props

If user sets `linkable` to `true`, then the property `id` becomes mandatory

```suggestion
type Linkable = {
  id: string
  linkable: true
} | {  linkable?: false }
type TypographyParams = PropsWithChildren<TypographyProps & Linkable>
```

Reply:
Thank you for the suggestion! Implemented with this [commit](https://github.com/podkrepi-bg/frontend/pull/278/commits/2acb40725303920a7176445428506f95466b2249)